src/DocumentStore.zig

```diff
@@ -588,6 +600,32 @@ fn getOrLoadBuildFile(self: *DocumentStore, uri: Uri) ?*BuildFile {
     return gop.value_ptr.*;
 }
 
+pub fn doubleCheckPotentialBuildFileUri(self: *DocumentStore, handle: *Handle) error{OutOfMemory}!?Uri {
+    handle.impl.lock.lock();
+    defer handle.impl.lock.unlock();
+    switch (handle.potential_build_file_uris.items.len) {
+        0 => {},
+        else => |count| blk: {
+            _ = self.getOrLoadBuildFile(handle.potential_build_file_uris.items[0]) orelse break :blk;
+            handle.associated_build_file = handle.potential_build_file_uris.items[0];
+            if (count == 1) break :blk;
+            const ok: bool = ok: for (handle.potential_build_file_uris.items[1..], 1..) |build_file_uri, index| {
```

Comment:
How about have a loop with the following structure?
`handle.associated_build_file = for (handle.potential_build_file_uris.items) { ... };`

You can have a special check before for `handle.potential_build_file_uris.items.len == 1`. This should avoid having the boolean to reset the associated_build_file after the loop.